File: js/shortest_word.js
function findShort(s){
  // split string into array
  const wordArr = s.split(' ')
  let smallestWord = wordArr[0]
  // loop through array
  for (let i = 0; i < wordArr.length; i++) {
  // if length of current word is less than previous word,
    if (wordArr[i].length < smallestWord.length) {
    // let smallestWord = current word
    smallestWord = wordArr[i]
    }
  }
  return smallestWord.length
}

File: java/shortest_word.java
import java.util.Arrays;

public class Kata {
    public static int findShort(String s) {
      // split string into array
      String[] arr = s.split(" ");
      // set the smallest word to be the first word in array
      String smallestWord = arr[0];

      for (int i = 0; i < arr.length; i++) {
        // if the length of the current word is less than smallestWord
        if (arr[i].length() < smallestWord.length()) {
          // let smallestWord = current word
          smallestWord = arr[i];
        }
      }
      // return length of smallestWord
      return smallestWord.length();
    }
}
